fsm/fsm.go

```diff
@@ -30,12 +33,25 @@ type FSM struct {
 	transitions   _TransitionMap
 }
 
+// State interface specifies the required interface for a State in the FSM.
 type State interface {
+	// ID is a unique numeric ID representing a State
 	ID() int
 	MarshalZerologObject(e *zerolog.Event)
+	// Name is the human-friendly name of the State
 	Name() string
 }
 
+// Transition is a user-specified Transition.  From is the old state.  To is the
+// new State.  Guards are a collection of zero or more callbacks that are
+// executed to determine if a transition may occur.  If a Guard returns an
+// error, the transition will fail.  Once all GlobalGuards and
+// Transition-specific Guards complete successfully, any OnEnterToActions will
+// be executed when the To state is being entered (error handling must be
+// handled by the caller and any error conditions should be detected in a
+// Guard).  Any OnExitToActions will be called when the FSM transitions to a
+// different state (again, error handling must be handled in a Guard or by the
+// handler itself).
```

Comment:
The order of this paragraph makes it sound like Enter callbacks will be invoked before Exit callbacks. The inverse is true.